Mathlib/Algebra/GeomSum.lean

```diff
@@ -191,6 +191,11 @@ theorem nat_sub_dvd_pow_sub_pow (x y n : ℕ) : x - y ∣ x ^ n - y ^ n := by
   · have : x ^ n ≤ y ^ n := Nat.pow_le_pow_left h.le _
     exact (Nat.sub_eq_zero_of_le this).symm ▸ dvd_zero (x - y)
 
+lemma nat_pow_sub_dvd_pow_mul_sub (x m n : ℕ) : x ^ m - 1 ∣ x ^ (m * n) - 1 := by
```

Comment:
I'm confused why we're writing this at all. It follows, in more generality (i.e., `Ring` instead of `Nat`), immediately from `sub_one_dvd_pow_sub_one` by using `x := x ^ m` (and `pow_mul`)

Reply:
Why not remove all the simple lemmas, and let not-quite-top mathematicians (and AI) struggle and waste their time to derive them when they need them, inventing the wheel over and over again? 

I'm done with "oh and if you come upon a lemma that's missing just submit a PR".

Sorry for the rant.

Reply:
Sorry, point taken. But still, the generality you want is `Ring`. The fact that it happens to hold for `Nat` is essentially a coincidence. (Because `Nat` subtraction is awful, and it just so happens that either, (a) it's zero, and zero divide anything, it (b) it's nonzero, in which case subtracting one is still well-behaved.)

That said, if you need this lemma for `Nat`, it's fine to add that special case.

Reply:
See the latest comment in the PR main comment thread.